core/tx_pool.go

```diff
@@ -648,6 +660,10 @@ func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 		log.Warn("tx has insufficient gas", "gas supplied", tx.Gas(), "gas needed", intrGas, "tx", tx)
 		return ErrIntrinsicGas
 	}
+	if len(pool.sendersCh) == 1000 {
+		log.Error("sendersCh is full, blocking until there is room")
+	}
+	pool.sendersCh <- newSender{tx.Hash(), internal}
```

Comment:
dont we want to use `select` here ? because we dont want to block the validateTx function if sendersCh is full or something

Reply:
Fixed